src/App.jsx

```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ALPHABETICALLY = 'alphabetically';
+
+function getPreperedGoods(goods, sortGoods, isReversedGoods) {
+  const preperedGoods = [...goods];
+
+  if (sortGoods) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortGoods) {
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+        case SORT_BY_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversedGoods) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversedGoods, setIsReversedGoods] = useState(false);
+  const sortedGoods = getPreperedGoods(
+    goodsFromServer,
+    sortField,
+    isReversedGoods,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isSortByLength = sortField !== SORT_BY_LENGTH;
+  const isSortByName = sortField !== SORT_BY_ALPHABETICALLY;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': isSortByName },
+          )}
+          onClick={() => setSortField(SORT_BY_ALPHABETICALLY)}
```

Comment:
(Optional) This can be moved to a separate handler and improve code readability.